### ISSUE TITLE
Illegal Access on protected variables from inherited classes

### ISSUE THREAD

**Issue by zekronium** [vsilaev/tascalate-async-await#8]
When a method is annotated with `@async` it seems to fail with with such error when trying to access a protected method

```
java.lang.IllegalAccessError: class DelayedActor$3 tried to access protected field io.vertx.core.AbstractVerticle.vertx (DelayedActor$3 and io.vertx.core.AbstractVerticle are in unnamed module of loader 'app')
```

This happens with any protected variable but protected methods work ok.

The error ***does not*** occur when parent class is in the same package as subclass only when its in a different package

Replacing some declared methods is possible but others are out of our control in most cases. Is this a bug or invalid use?

```java
public class DelayedActor extends Actor {
    @Override
    @async
    void run() {
        try {
            await(iter());
        }catch (Throwable e){
            //just here to print the error
            e.printStackTrace();
        }
    }

    private @async CompletionStage<String> iter(){
        for (int i = 0; i < 100_000; i++) {
            await(sleep(1000));
        }
        return async("done");
    }

    private  @async CompletionStage<Void> sleep(long time){
        final CompletableFuture<Void> callback = new CompletableFuture<>();
        //here super.vertx is an illegal access
        super.vertx.setTimer(time, t -> callback.complete(null));
        return callback;
    }
}

public abstract class Actor extends AbstractVerticle {

    @Override
    public void start() throws Exception {
        this.run();
    }

    abstract void run();
}
```

Weirdly enough, removing the `@async` annotation on sleep works completely fine and behaves as expected. Is `@async` annotation only required when you want to `await` in the method?

**Comment by vsilaev** [comment]
> This happens with any protected variable but protected methods work ok.
> 
> The error _**does not**_ occur when parent class is in the same package as subclass only when its in a different package

Yes, this sounds like an error on my side - overlooked visibility rules for inheritance.



> Weirdly enough, removing the `@async` annotation on sleep works completely fine and behaves as expected. Is `@async` annotation only required when you want to `await` in the method?

Usage of `@async` is wrong here: you must use `return async(value)` construct to return `value` from the async method. And yes, `@async` is completely unnecessary in your example. It must be used if you are calling `await` or calling `@suspendable` methods or working with generators (the latest is undocumented yet).

**Comment by zekronium** [comment]
Was this fixed in latest version?

**Comment by vsilaev** [comment]
Fixed in release [1.2.2](https://github.com/vsilaev/tascalate-async-await/releases/tag/1.2.2)
Please verify and close the issue

**Comment by zekronium** [comment]
Fixed
